FV3/gfsphysics/GFS_layer/GFS_physics_driver.F90

```diff
@@ -5677,7 +5677,38 @@ subroutine moist_bud2(im,ix,ix2,levs,me,kdt,grav,dtp,delp,rain, &
 
       end subroutine moist_bud2
 
-      subroutine moist_cv_nwat6(initial_dynamics_q, physics_q, pressure_on_interfaces, &
+      subroutine compute_q_liquid(im, levs, nwat, ntcw, ntrw, new_dynamics_q, q_liquid)
```

Comment:
This is a very clever way to get around the problem, it looks like this covers almost all the cases in `moist_cv` and `moist_cp` in `fv_mapz`, except for the special case in (2).

Reply:
That's right -- thanks for bringing that up.  I initially thought I could do without that special case, but on another inspection it looks like that version is used on occasion:

https://github.com/VulcanClimateModeling/fv3gfs-fortran/blob/201f4cc6da6a90a477e59d04f58f05f503317e6c/FV3/atmos_cubed_sphere/model/fv_update_phys.F90#L448-L449

I will see if I can revise things a bit tomorrow.

Reply:
I updated things to include the special case.  It improves the RMSE metric by another order of magnitude -- see the [updated notebook](https://github.com/VulcanClimateModeling/explore/blob/master/spencerc/2021-09-14-test-moist-cv/2021-09-14-test-moist-cv.ipynb) -- so it was definitely good to include it.

Reply:
Nice!